src/sagemaker/workflow/model_step.py

```diff
@@ -0,0 +1,198 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""The `ModelStep` definition for SageMaker Pipelines Workflows"""
+from __future__ import absolute_import
+
+import logging
+from typing import Union, List, Dict, Optional
+
+from sagemaker import Model, PipelineModel
+from sagemaker.workflow._utils import _RegisterModelStep, _RepackModelStep
+from sagemaker.workflow.pipeline_context import PipelineSession, _ModelStepArguments
+from sagemaker.workflow.retry import RetryPolicy
+from sagemaker.workflow.step_collections import StepCollection
+from sagemaker.workflow.steps import Step, CreateModelStep
+
+NEED_RUNTIME_REPACK = "need_runtime_repack"
+
+_CREATE_MODEL_RETRY_POLICIES = "create_model_retry_policies"
+_REGISTER_MODEL_RETRY_POLICIES = "register_model_retry_policies"
+_REPACK_MODEL_RETRY_POLICIES = "repack_model_retry_policies"
+_REGISTER_MODEL_NAME_BASE = "RegisterModel"
+_CREATE_MODEL_NAME_BASE = "CreateModel"
+_REPACK_MODEL_NAME_BASE = "RepackModel"
+
+
+class ModelStep(StepCollection):
+    """`ModelStep` for SageMaker Pipelines Workflows."""
+
+    def __init__(
```

Comment:
I don't think it builds the properties here. 

See: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/steps.py#L413

Reply:
No it does not
Synced up offline. We should append a separate PR to add model step properties reference and depended on by other steps as this PR is large enough.